src/mono/mono/metadata/class-internals.h

```diff
@@ -504,11 +504,11 @@ struct _MonoGenericContainer {
 	int type_argc    : 29; // Per the ECMA spec, this value is capped at 16 bits
 	/* If true, we're a generic method, otherwise a generic type definition. */
 	/* Invariant: parent != NULL => is_method */
-	int is_method     : 1;
+	gint is_method     : 1;
 	/* If true, this container has no associated class/method and only the image is known. This can happen:
 	   1. For the special anonymous containers kept by MonoImage.
 	   2. When user code creates a generic parameter via SRE, but has not yet set an owner. */
-	int is_anonymous : 1;
+	gint is_anonymous : 1;
```

Comment:
@am11 these still raise the warning in main, should this have been `guint`?

```
/__w/1/s/src/mono/mono/metadata/sre.c:1626:33: warning: implicit truncation from 'int' to a one-bit wide bit-field changes value from 1 to -1 [-Wsingle-bit-bitfield-constant-conversion]
                          generic_container->is_method = TRUE;
                                                       ^ ~~~~
  /__w/1/s/src/mono/mono/metadata/sre.c:1631:36: warning: implicit truncation from 'int' to a one-bit wide bit-field changes value from 1 to -1 [-Wsingle-bit-bitfield-constant-conversion]
                          generic_container->is_anonymous = TRUE;
                                                          ^ ~~~~
```

Reply:
https://github.com/dotnet/runtime/pull/91314

Reply:
Thanks, yes guint is correct.